### PR TITLE
feat: account variable set

### PR DESCRIPTION
**Pull request by jorrite**
This PR:
- Creates an account variable set.
- Creates the (at most) 3 variables (account, environment, workload_permissions_boundary_arn) inside the variable set, instead of passing them down to each workspace (and then creating individual workspace variables).
- Attach account variable set to all workspaces. 

The net result here is a lot of less resources. With the following example: 20 workspaces, 4 environments (accounts), 4 variables:

|   | variables | variable sets  | variable set attachments  |  total resources |
|---|---|---|---|---|
| before   | 320 [20×4×4]  |   |   |  320 | 
| after   |  16 [4×4]  | 4  | 80 [4x20]  | 100 |  

Additionally, variables and an output is provider to add more variables to the same account variable set. The variables are there for things you know up front, the output is there for variables that might have values that relate to things you provision _inside_ the accounts.

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -30,6 +30,42 @@ variable "account" {
...
+  default     = {}
+  description = ""
+}
+
+variable "account_variable_set_sensitive_terraform_variables" {
+  type        = map(string)
+  default     = {}
+  description = ""
+}
+
```
Comment (gitbluf) [inline variables.tf:68]: You should probably add description and not leave it empty.

**Comment by jorrite** [comment]
Updated the OP to reflect that this PR isn't a hybrid anymore, it changes the default behavior. ☝️.

File: main.tf [hunk 2]
```diff
@@ -202,6 +244,7 @@ module "additional_tfe_workspaces" {
   terraform_version              = each.value.terraform_version != null ? each.value.terraform_version : var.tfe_workspace.terraform_version
   trigger_prefixes               = each.value.connect_vcs_repo != false ? coalesce(each.value.trigger_prefixes, var.tfe_workspace.trigger_prefixes) : null
   username                       = coalesce(each.value.username, "TFEPipeline-${each.key}")
+  variable_set_ids               = { "account" : tfe_variable_set.account.id }
```
Comment (marwinbaumannsbp) [inline main.tf:247]: you need to merge this with each.value.variable_set_ids

File: main.tf [hunk 3]
```diff
@@ -159,13 +200,14 @@ module "tfe_workspace" {
   terraform_version              = var.tfe_workspace.terraform_version
   trigger_prefixes               = var.tfe_workspace.connect_vcs_repo != false ? var.tfe_workspace.trigger_prefixes : null
   username                       = var.tfe_workspace.username
+  variable_set_ids               = { "account" : tfe_variable_set.account.id }
```
Comment (marwinbaumannsbp) [inline main.tf:203]: you need to merge this with var.tfe_workspace.variable_set_ids

File: main.tf [hunk 4]
```diff
@@ -1,5 +1,7 @@
 locals {
-  tfe_workspace = {
+  account_variable_set = {
+    name = var.account_variable_set.name != null ? var.account_variable_set.name : var.name
```
Comment (marwinbaumannsbp) [inline main.tf:3]: var.name -> workload-${var.name} or account-${var.name}? To differentiate with e.g. the inception variable sets

File: main.tf (hunk 2 above)
Comment (jorrite) [reply main.tf:247]: Ah, yes!

File: main.tf (hunk 4 above)
Comment (jorrite) [reply main.tf]: Done.

File: main.tf (hunk 3 above)
Comment (jorrite) [reply main.tf]: Indeed, fixed.

File: main.tf (hunk 2 above)
Comment (jorrite) [reply main.tf]: Fixed.

**Review by marwinbaumannsbp — APPROVED** [review]

**Pull request merged**